File: Deepseek-Coder-33b-Instruct/stage2/data_construct_copyout_clause.cpp
#include <stdlib.h>
#include <openacc.h>

#ifndef T1
//T1:data construct copyout clause,V:2.7-3.3
int test1(){
    int err = 0;
    const int N = 100;
    int *host_data = (int*)malloc(N * sizeof(int));
    int *device_data;

    // Initialize host data
    for(int i = 0; i < N; i++) {
        host_data[i] = i;
    }

    // Allocate device memory
    device_data = (int*)acc_malloc(N * sizeof(int));

    // Copy data to device
    #pragma acc data copyin(host_data[:N]) copyout(device_data[:N])
    {
        // Perform computation on device
        #pragma acc kernels
        for(int i = 0; i < N; i++) {
            device_data[i] = host_data[i] * 2;
        }
    }

    // Check if data was correctly copied back to host
    for(int i = 0; i < N; i++) {
        if(device_data[i] != host_data[i] * 2) {
            err = 1;
            break;
        }
    }

    // Free device memory
    acc_free(device_data);
    free(host_data);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/data_construct_copyout_clause.f90
!T1:data construct copyout clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        INTEGER :: i, host_array(N), device_array(N)

        ! Initialize host array
        DO i = 1, N
          host_array(i) = i
        END DO

        ! Copy host array to device
        !$acc data copyin(host_array) copyout(device_array)

        ! Copy host array to device array
        !$acc parallel loop present(host_array, device_array)
        DO i = 1, N
          device_array(i) = host_array(i)
        END DO

        ! Modify device array
        !$acc parallel loop present(device_array)
        DO i = 1, N
          device_array(i) = device_array(i) * 2
        END DO

        !$acc end data

        ! Copy device array back to host
        !$acc data copyin(device_array) copyout(host_array)

        ! Copy device array to host array
        !$acc parallel loop present(device_array, host_array)
        DO i = 1, N
          host_array(i) = device_array(i)
        END DO

        !$acc end data

        ! Check if host array was updated correctly
        DO i = 1, N
          IF (host_array(i) .ne. i * 2) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM